nemo/core/classes/modelPT.py

```diff
@@ -193,8 +197,74 @@ def save_to(self, save_path: str):
             torch.save(self.state_dict(), model_weights)
             self.__make_nemo_file_from_folder(filename=save_path, source_dir=tmpdir)
 
+    def _eff_save_to(self, save_path: str):
+        """
+        Saves model instance (weights and configuration) into an EFF archive.
+
+        EFF file is an archive (tar.gz) with the following:
+            metadata.yaml - file describing the content of the archive.
+            model_config.yaml - model configuration in .yaml format. 
+                You can deserialize this into cfg argument for model's constructor
+            model_wights.chpt - model checkpoint
+
+        Note that for NeMo the EFF archives will use .nemo postfix.
+
+        Args:
+            save_path: Path to archive file where model instance should be saved.
+        """
+        # Create EFF archive.
+        with Archive.create(save_path=save_path, runtime=Runtimes.PyTorch, origin=Origins.NeMo) as effa:
+
+            # Add config file to archive.
+            config_yaml = effa.create_file_handle(
+                name=_MODEL_CONFIG_YAML, description="File containing model configuration"
+            )
+            self.to_config_file(path2yaml_file=config_yaml)
+
+            # Add model weights to archive.
+            model_weights = effa.create_file_handle(name=_MODEL_WEIGHTS, description="File containing model weights")
+            torch.save(self.state_dict(), model_weights)
+
+            # Add other artifacts to archive.
+            if hasattr(self, 'artifacts') and self.artifacts is not None:
+                # Iterate through artifact one by one.
+                for (conf_path, src) in self.artifacts:
+                    try:
+                        # Add artifact - save `conf_path` as description.
+                        _ = effa.create_file_handle(name=src, description=conf_path)
+                        # Copy file
+                        if os.path.exists(src):
+                            shutil.copy2(src, effa.tmpdir)
+                    except Exception:
+                        logging.error(f"Could not copy artifact {src} used in {conf_path}")
+
+    @rank_zero_only
+    def save_to(self, save_path: str):
+        """
+        Saves model instance (weights and configuration) into EFF archive or .
+         You can use "restore_from" method to fully
+        restore instance from .nemo file.
+
+        .nemo file is an archive (tar.gz) with the following:
+            model_config.yaml - model configuration in .yaml format. You can deserialize this into cfg argument for model's constructor
+            model_wights.chpt - model checkpoint
+
+        Args:
+            save_path: Path to .nemo file where model instance should be saved
+        """
+        # Add nemo rank check as well
+        if not is_global_rank_zero():
+            return
+
+        if _EFF_PRESENT_:
```

Comment:
Such a global flag is not a good design - what if a user has eff installed but what's nemo checkpoints (when not working with TLT workflow). Create a global variable, create a setter method to enable / disable eff type file saving, and use that flag here.

Reply:
"what if a user has eff installed but what's nemo checkpoints" - not sure what you mean by that, can you please explain?

Reply:
> Create a global variable, create a setter method to enable / disable eff type file saving, and use that flag here.

Not sure what you mean by "setter" method... who and when should call it?

Can you please take a look at `save_to` and `restore_from` and tell me what is missing?

Reply:
Or more generally: what do you think on the logic of `save_to` and `restore_from` provided in the description? What is missing?

Reply:
Ok, we discussed that with @titu1994, and the comment was not about bad "global flag" design (which is commonly used in NeMo for handling other conditional dependencies), but rather the lack of option for programmatically enabling/disabling using EFF.

I honestly see low usability of this, but added additional _MODEL_EFF_SAVE flag with setter/getter static methods to model, so the user, in his code, will be able to manually turn off saving .nemo files with EFF.

Reply:
Yep, poor wording on my part. Thanks for his Tom

Reply:
The opposite - thanks for reviewing and discussion.